py/tests/network.py

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# module for getting the lan ip address of the computer
+import os
+import socket
+
+if os.name != "nt":
+    import fcntl
+    import struct
+    def get_interface_ip(ifname):
+        sckt = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        return socket.inet_ntoa(fcntl.ioctl(
+                sckt.fileno(),
+                0x8915,  # SIOCGIFADDR
+                struct.pack('256s', ifname[:15])
+            )[20:24])
+
+def get_lan_ip():
+    try:
+        ip = socket.gethostbyname(socket.gethostname())
+    except:
+        return '0.0.0.0'
+    if ip.startswith("127.") and os.name != "nt":
+        interfaces = ["eth0","eth1","eth2","en0","en1","en2","en3","en4","wlan0","wlan1","wifi0","ath0","ath1","ppp0"]
```

Comment:
This will fail on most newer linux hosts.
You can't rely on a hardcoded list of possible interface names.  Now that most linux distros are using systemd, the traditional interface naming scheme isn't used anymore.

For example, on my machine (Ubuntu 16.04), the network interface for my ethernet adapter is assigned "enp4s0".  For the gory details, see: https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/

One possible solution would be to inspect `/sys/class/net` to get the local interface name (however, that will only work on Linux, not on other systems like mac osx).